lib/src/http_headers.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:ffi';
+
+import 'package:ffi/ffi.dart';
+
+import 'globals.dart';
+import 'third_party/cronet/generated_bindings.dart';
+
+/// Headers for HTTP requests.
+///
+/// In some situations, headers are immutable:
+/// * [HttpClientRequest] have immutable headers from the moment the body is
+/// written to.
```

Comment:
Add the rest of the documentation from https://api.dart.dev/stable/2.13.4/dart-io/HttpHeaders-class.html as well.

Reply:
Added some more. Can't add all of it as in this PR, not all of the HttpHeaders apis are implemented.